src/gateway/GatewayV1.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "./Gateway.sol";
+
+interface IBalancerFlashLoan {
+    function flashLoan(
+        address recipient,
+        IERC20[] memory tokens,
+        uint256[] memory amounts,
+        bytes memory userData
+    ) external;
+}
+
+/// @title ECG Gateway V1
+/// @notice Gateway to interract via multicall with the ECG
+/// Owner can select which calls are allowed
+/// @custom:feature flashloan from balancer vault
+contract GatewayV1 is Gateway {
+    /// @notice Address of the Balancer Vault, used for initiating flash loans.
+    address public immutable balancerVault =
+        0xBA12222222228d8Ba445958a75a0704d566BF2C8;
+
+    function multicallWithBalancerFlashLoan(
+        IERC20[] calldata tokens,
+        uint256[] calldata amounts,
+        bytes[] calldata calls // Calls to be made after receiving the flash loan
+    ) public whenNotPaused {
+        require(
+            _originalSender == address(1),
+            "Gateway: original sender already set"
+        );
+
+        _originalSender = msg.sender;
+        // Clear existing StoredCalls
+        delete _storedCalls;
+
+        // Manually copy each element
+        for (uint i = 0; i < calls.length; i++) {
+            _storedCalls.push(calls[i]);
+        }
```

Comment:
todo: check if this can be optimized (e.g. set storage to 0xfffffff instead of 0x0000 to avoid re-initialization of state)